sklearn/isotonic.py

```diff
@@ -118,7 +118,7 @@ def isotonic_regression(
     order = np.s_[:] if increasing else np.s_[::-1]
     y = check_array(y, ensure_2d=False, dtype=[np.float64, np.float32])
     y = np.array(y[order], dtype=y.dtype)
-    sample_weight = _check_sample_weight(sample_weight, y, dtype=y.dtype)
+    sample_weight = _check_sample_weight(sample_weight, y, dtype=y.dtype, copy=True)
```

Comment:
Could you make a similar change in l.267 for the `IsotonicRegression._build_y` function?
We got lucky because we trigger a copy at some point but it might be error-prone as well.

Reply:
I do not think we need to copy on L273, because `_make_unique` creates a new `unique_sample_weight` and `sample_weight` is not used anymore afterwards.

Reply:
OK. At least we will have the non-regression test if this is changing in the futurte.